include/circt/Dialect/Moore/MooreOps.td

```diff
@@ -454,6 +454,21 @@ def NamedConstantOp : MooreOp<"named_constant", [
   }];
 }
 
+def StringConstantOp : MooreOp<"string", [Pure, ConstantLike]> {
+  let summary = "Produce a constant string value";
+  let description = [{
+    Produces a constant value of string type.
+
+    Example:
+    ```mlir
+    %0 = moore.string "hello world"
+    ```
+  }];
+  let arguments = (ins StrAttr:$value);
+  let results = (outs SimpleBitVectorType:$result);
```

Comment:
Can we restrict this to a StringType?

Reply:
Maybe not. 

Here is the sv content
```
module test();
    string s="hello world";
endmodule
```


And slang ast json file generated below.
```
{
  "name": "$root",
  "kind": "Root",
  "addr": 100747055908128,
  "members": [
    {
      "name": "",
      "kind": "CompilationUnit",
      "addr": 100747056022832
    },
    {
      "name": "test",
      "kind": "Instance",
      "addr": 100747056023696,
      "body": {
        "name": "test",
        "kind": "InstanceBody",
        "addr": 100747056023104,
        "members": [
          {
            "name": "s",
            "kind": "Variable",
            "addr": 100747056023352,
            "type": "string",
            "initializer": {
              "kind": "Conversion",
              "type": "string",
              "operand": {
                "kind": "StringLiteral",
                "type": "bit[87:0]",
                "literal": "hello world",
                "constant": "88'h68656c6c6f20776f726c64"
              },
              "constant": "\"hello world\""
            },
            "lifetime": "Static"
          }
        ],
        "definition": "test"
      },
      "connections": [
      ]
    }
  ]
}
```
I reckon there is a Conversion between bits and string.

Reply:
Hummm...It looks like slang turns the string into the bits array.

Reply:
Ok I see. In that case it looks reasonable to treat string literal as a packed bit.

Reply:
Oh that's a nice find! So it looks like string literals are basically just a different way to create an integer constant?